nixos/modules/services/mail/listmonk.nix

```diff
@@ -0,0 +1,212 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.listmonk;
+  tomlFormat = pkgs.formats.toml {};
+  cfgFile = tomlFormat.generate "listmonk.toml" cfg.settings;
+  setDatabaseOption = key: value: "UPDATE settings SET value = '${builtins.toJSON value}' WHERE key = '${key}';";
+  updateDatabaseConfigSQL = pkgs.writeText "update-database-config.sql"
+    (concatStringsSep "\n" (mapAttrsToList setDatabaseOption (if (cfg.databaseSettings != null) then cfg.databaseSettings else {})));
+  updateDatabaseConfigScript = pkgs.writeShellScriptBin "update-database-config.sh" ''
+    ${if cfg.mutableDatabaseSettings then ''
+      if [ ! -f $STATE_DIRECTORY/.db_settings_initialized ]; then
+        ${pkgs.postgresql}/bin/psql -d listmonk -f ${updateDatabaseConfigSQL} ;
+        touch $STATE_DIRECTORY/.db_settings_initialized
+      fi
+    '' else "${pkgs.postgresql}/bin/psql -d listmonk -f ${updateDatabaseConfigSQL}"
+    }
+  '';
+in
+{
+  ###### interface
+  options = {
+    services.listmonk = {
+      enable = mkEnableOption "Listmonk";
```

Comment:
You should mention the assumption is that this will sit behind a reverse proxy. If upstream provides any documentation for configuring a reverse proxy it would be nice to include that too.

Reply:
I do not think there is doc from upstream, but I added it assumes a reverse proxy.